ui/components/src/TimeSeriesTooltip/TooltipContent.tsx

```diff
@@ -12,107 +12,89 @@
-  const { formatWithUserTimeZone } = useTimeZone();
+  const { series, wrapLabels } = props;
 
-  const seriesTime = series && series[0] && series[0].date ? series[0].date : null;
-
-  const formatTimeSeriesHeader = (timeMs: number) => {
-    const date = new Date(timeMs);
-    const formattedDate = formatWithUserTimeZone(date, 'MMM dd, yyyy - ');
-    const formattedTime = formatWithUserTimeZone(date, 'HH:mm:ss');
-    return (
-      <Box>
-        <Typography
-          variant="caption"
-          sx={(theme) => ({
-            color: theme.palette.common.white,
-          })}
-        >
-          {formattedDate}
-        </Typography>
-        <Typography variant="caption">
-          <strong>{formattedTime}</strong>
-        </Typography>
-      </Box>
-    );
-  };
+  const theme = useTheme();
 
   const sortedFocusedSeries = useMemo(() => {
     if (series === null) return null;
     return series.sort((a, b) => (a.y > b.y ? -1 : 1));
   }, [series]);
 
-  if (sortedFocusedSeries === null || seriesTime === null) {
+  if (series === null || sortedFocusedSeries === null) {
     return null;
   }
+  const totalSeries = series.length;
 
-  // TODO: use react-virtuoso to improve performance
-  return (
-    <Stack py={1} spacing={0.5}>
-      <Box
-        sx={{
-          display: 'flex',
-          justifyContent: 'start',
-          alignItems: 'center',
-        }}
-      >
-        {formatTimeSeriesHeader(seriesTime)}
-        <Stack direction="row" gap={1} sx={{ marginLeft: 'auto' }}>
-          <Typography sx={{ fontSize: 11 }}>Click to {isTooltipPinned ? 'Unpin' : 'Pin'}</Typography>
-          {isTooltipPinned ? (
-            <Pin onClick={onUnpinClick} sx={{ fontSize: 16, cursor: 'pointer' }} />
-          ) : (
-            <PinOutline sx={{ fontSize: 16 }} />
-          )}
-        </Stack>
-      </Box>
-
-      <Divider
-        sx={(theme) => ({
-          borderColor: theme.palette.grey['500'],
-        })}
+  if (totalSeries === 1) {
+    const [seriesData] = series;
+    if (seriesData === undefined) {
+      return null;
+    }
+    const { seriesName, y, formattedY, markerColor, isClosestToCursor } = seriesData;
+    return (
+      <SeriesInfo
+        seriesName={seriesName}
+        y={y}
+        formattedY={formattedY}
+        markerColor={markerColor}
+        totalSeries={sortedFocusedSeries.length}
+        wrapLabels={wrapLabels}
+        emphasizeText={isClosestToCursor}
       />
-      <Box
-        sx={{
-          display: 'table',
-        }}
-      >
-        {sortedFocusedSeries.map(
-          ({ datumIdx, seriesIdx, seriesName, y, formattedY, markerColor, isClosestToCursor }) => {
-            if (datumIdx === null || seriesIdx === null) return null;
-            const key = seriesIdx.toString() + datumIdx.toString();
+    );
+  }
+
+  // TODO: is there a better way to approximate height or a dynamic height and width workaround for Virtuoso?
```

Comment:
@juliepagano this is my first time using react-virtuoso and I'm definitely not doing everything correctly. I tried following some of their examples and the code you wrote for the legend is really helpful, but curious if you can give me some pointers? 🙏 

Currently, I'm not sure the best way to set the `style={{ height: contentHeight, width: TOOLTIP_MULTI_SERIES_MIN_WIDTH }}` lines below. I'm approximating based on the number of series, which kind of works for longer series names, but results in weirdness like here:
<img width="777" alt="image" src="https://github.com/perses/perses/assets/9369625/47059d75-c60b-45da-9c7f-bcc792364373">

I thought about improving the approximation by checking the number of characters in each series name to find the longest (and use that to infer how many lines it wraps to), but that seems hacky and bad for performance. I did try some workarounds after Googling "react-virtuoso dynamic height and width" but they didn't seem to work, it does seem like the outer container needs a static height defined and since this changes as you move your cursor when number of nearby series changes, I'm not how to address this 😕

Reply:
Figured out an approach that works with useResizeObserver, closing this PR since I’m going to split it up into multiple PRs so it’s easier to review